File: leetcode/src/test/java/com/giwankim/leetcode/ValidParenthesesTest.java
package com.giwankim.leetcode;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

class ValidParenthesesTest {
  ValidParentheses sut = new ValidParentheses();

  @ParameterizedTest
  @CsvSource({"(),true", "()[]{},true", "(],false", "([]),true", "],false"})
  void isValid(String s, boolean expected) {
    boolean actual = sut.isValid(s);
    assertThat(actual).isEqualTo(expected);
  }
}

File: leetcode-kt/src/test/kotlin/com/giwankim/leetcode/ValidParenthesesTest.kt
package com.giwankim.leetcode

import io.kotest.core.spec.style.FunSpec
import io.kotest.datatest.withTests
import io.kotest.matchers.shouldBe

class ValidParenthesesTest :
    FunSpec({
        val sut = ValidParentheses()

        context("valid parentheses") {
            withTests(
                nameFn = { (s, expected) -> "s=$s, expected=$expected" },
                ValidParenthesesCase("()", true),
                ValidParenthesesCase("()[]{}", true),
                ValidParenthesesCase("(]", false),
                ValidParenthesesCase("([])", true),
                ValidParenthesesCase("]", false),
            ) { (s, expected) ->
                sut.isValid(s) shouldBe expected
            }
        }
    })

private data class ValidParenthesesCase(
    val s: String,
    val expected: Boolean,
)
